From 3c774fb401cd8cafe76f4611648bdd7e44817be0 Mon Sep 17 00:00:00 2001
From: Jiri Popelka <jpopelka@redhat.com>
Date: Mon, 15 Sep 2014 16:18:23 +0200
Subject: amend previous commit


diff --git a/printerproperties.py b/printerproperties.py
index 41b12c2..f4da9f4 100755
--- a/printerproperties.py
+++ b/printerproperties.py
@@ -1333,7 +1333,7 @@ class PrinterPropertiesDialog(GtkGUI):
             # Either the underlying cupsGetPPD2() function returned
             # NULL without setting an IPP error (so it'll be something
             # like a failed connection), or the PPD could not be parsed.
-            if str (e).startswith ("ppd")
+            if str (e).startswith ("ppd"):
                 show_error_dialog (_("Error"),
                                    _("The PPD file for this queue "
                                      "is damaged."),
-- 
cgit v0.10.2

