From d2fc84028f0df021226a1dd7a84d7227daff16a6 Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Fri, 5 Sep 2014 12:16:49 +0100
Subject: Take the gdk lock before entering gtk_main() (bug #1052203 comment
 24).


diff --git a/scp-dbus-service.py b/scp-dbus-service.py
index 76f7056..945e53a 100644
--- a/scp-dbus-service.py
+++ b/scp-dbus-service.py
@@ -2,7 +2,7 @@
 
 ## system-config-printer
 
-## Copyright (C) 2010, 2011, 2012, 2013 Red Hat, Inc.
+## Copyright (C) 2010, 2011, 2012, 2013, 2014 Red Hat, Inc.
 ## Authors:
 ##  Tim Waugh <twaugh@redhat.com>
 
@@ -573,5 +573,7 @@ if __name__ == '__main__':
     debugprint ("Service running...")
     g_killtimer = KillTimer (killfunc=Gtk.main_quit)
     cp = ConfigPrinting ()
+    Gdk.threads_enter ()
     Gtk.main ()
+    Gdk.threads_leave ()
     cp.destroy ()
-- 
cgit v0.10.2

