diff -Naur linux-3.16.2.ORIG/tools/perf/util/util.h linux-3.16.2/tools/perf/util/util.h
--- linux-3.16.2.ORIG/tools/perf/util/util.h	2014-09-06 01:37:11.000000000 +0200
+++ linux-3.16.2/tools/perf/util/util.h	2014-09-15 21:50:22.178148831 +0200
@@ -38,7 +38,7 @@
 #define decimal_length(x)	((int)(sizeof(x) * 2.56 + 0.5) + 1)
 
 #define _ALL_SOURCE 1
-#define _BSD_SOURCE 1
+#define _DEFAULT_SOURCE 1
 #define HAS_BOOL
 
 #include <unistd.h>
